configuration/filesystem/nfs-server.sh

```diff
@@ -0,0 +1,268 @@
+# shared-storage PS, and that the values that we're picking up from trinity.sh
+# are sane...
+
+if flag_is_unset HA ; then
+    echo_info 'Non-HA setup, disabling NFS_EXPORT_LOCAL and NFS_EXPORT_IMAGES'
+    read NFS_EXPORT_{LOCAL,IMAGES} <<< "0 0"
+fi
+
+
+case $SHARED_FS_TYPE in
+
+    none )
+        echo_info '"None" use case selected, not setting up the NFS server.'
+        read NFS_EXPORT_{LOCAL,IMAGES,SHARED,HOME} <<< "0 0 0 0"
+        exit 0
+        ;;
+
+    export | dev | drbd )
+        NFS_EXPORT_LOCAL=${NFS_EXPORT_LOCAL:-1}
+        NFS_EXPORT_IMAGES=${NFS_EXPORT_IMAGES:-1}
+        NFS_EXPORT_SHARED=${NFS_EXPORT_SHARED:-1}
+        NFS_EXPORT_HOME=${NFS_EXPORT_HOME:-1}
+        ;;
+
+    * ) # We should never get there
+        echo_error 'Invalid shared storage use case, exiting.'
+        exit 1
+esac
+
+
+flag_is_set NFS_ENABLE_RDMA && proto=rdma || proto=tcp
+
+
+
+#---------------------------------------
+# Non-HA
+#---------------------------------------
+
+if flag_is_unset HA ; then
+
+    render_template "${POST_FILEDIR}"/sysconfig_nfs > /etc/sysconfig/nfs
+    setup_exports "${POST_FILEDIR}"/nonHA_exports /etc/exports.d/trinity.exports
+    sync
+    start_nfs_server
+    showmount -e
+
+
+
+#---------------------------------------
+# HA primary
+#---------------------------------------
+
+elif flag_is_set PRIMARY_INSTALL ; then
+
+    render_template "${POST_FILEDIR}"/sysconfig_nfs > /etc/sysconfig/nfs
+    setup_exports "${POST_FILEDIR}"/HA_exports "${TRIX_LOCAL}"/etc/exports.d/trinity.exports
+    symlink_exports
+    render_template "${POST_FILEDIR}"/nfsmount.conf > /etc/nfsmount.conf
+    sync
+
+    # Information that should survive a failover
+    mkdir -p "${TRIX_LOCAL}"/var/lib/nfs
+
+
+    # Let's see if the config flies
+    start_nfs_server
+    showmount -e
+    stop_nfs_server
+
+
+    echo_info 'Setting up the NFS server Pacemaker resource'
+
+    tmpfile=$(mktemp -p /root pacemaker_nfs.XXXX)
+    pcs cluster cib $tmpfile
+
+    pcs -f $tmpfile resource create trinity-nfs-server ocf:heartbeat:nfsserver \
+        nfs_shared_infodir="${TRIX_LOCAL}"/var/lib/nfs op monitor interval=47s
+
+    pcs -f $tmpfile resource group add Trinity trinity-nfs-server --after trinity-fs
+
+    # Apply the changes
+    if ! pcs cluster cib-push $tmpfile ; then
+        echo_error 'Failed to push the new resource configuration to Pacemaker, exiting.'
+        exit 1
+    fi
+
+
+
+#---------------------------------------
+# HA secondary
+#---------------------------------------
+
+else
+
+    render_template "${POST_FILEDIR}"/sysconfig_nfs > /etc/sysconfig/nfs
+    symlink_exports
+    render_template "${POST_FILEDIR}"/nfsmount.conf > /etc/nfsmount.conf
+    sync
+
+
+    echo_info 'Setting up the NFS Pacemaker mounts'
+
+    tmpfile=$(mktemp -p /root pacemaker_nfs-clients.XXXX)
+    pcs cluster cib $tmpfile
+
+    if flag_is_set NFS_EXPORT_LOCAL ; then
+
+        pcs -f $tmpfile resource create trinity-nfs-client-local \
```

Comment:
Wasn't it agreed that we will not mount resources on passive controller from active node?